ckanext/validation/utils.py

```diff
@@ -43,17 +43,13 @@ def process_schema_fields(data_dict):
     Note that the data_dict still needs to pass validation
     '''
 
+    schema_upload = data_dict.pop(u'schema_upload', None)
+    schema_url = data_dict.pop(u'schema_url', None)
     # Distinguish blank field (likely web UI) from absent field (likely API).
     # Absent fields do not overwrite the existing schema,
     # but fields that are present and blank can.
     has_schema_json = u'schema_json' in data_dict
 
-    schema_upload = data_dict.pop(u'schema_upload', None)
-    schema_url = data_dict.pop(u'schema_url', None)
-    schema_json = data_dict.pop(u'schema_json', None)
```

Comment:
should we still pop it outside of the loop so its not there if an upload or url also exists?